### PR TITLE
Fix result code for PrepareAudioPassThru

### PR DESCRIPTION
**Pull request by OlhaVorobiova**
Fixes #[3845](https://github.com/smartdevicelink/sdl_core/issues/3845)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
ATF scripts

### Summary
- Added a check success result codes in PrepareAudioPassThruResultCodeForResponse
- Added  a result code TRUNCATED_DATA in PerformAudioPassThruRequest::on_event 
### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_audio_pass_thru_request.cc [hunk 1]
```diff
@@ -198,7 +199,10 @@ void PerformAudioPassThruRequest::on_event(const event_engine::Event& event) {
     return;
   }
 
-  const ResponseParams response_params = PrepareResponseParameters();
+  ResponseParams response_params = PrepareResponseParameters();
+  if (message[strings::params].keyExists(hmi_response::message)) {
+    response_params.success = false;
```
Comment (JackLivio) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_audio_pass_thru_request.cc:204]: I am not sure this is the correct logic for handling if success is false. If the HMI sends a WARNINGS, SUCCESS, RETRY or the acceptable "success=true" related then success should be true. The HMI i believe can only send a message param with pertinent warning related messages if it is in the hmi error response structure.

Comment (JackLivio) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_audio_pass_thru_request.cc:204]: @OlhaVorobiova @dboltovskyi

Comment (dboltovskyi) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_audio_pass_thru_request.cc:204]: @VjKlepikov Could you support please

Comment (GetmanetsIrina) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_audio_pass_thru_request.cc]: @JackLivio, please find updates in [1f81d87](https://github.com/smartdevicelink/sdl_core/pull/3854/commits/1f81d87c448c4d5473a4fbd066355f70003f4242)

**Pull request merged**